core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java

```diff
@@ -327,7 +325,7 @@ private boolean tryRehash()
         values = newValues;
         groupIds = newGroupIds;
 
-        this.valuesByGroupId.ensureCapacity(maxFill);
+        this.valuesByGroupId = Arrays.copyOf(valuesByGroupId, maxFill);
```

Comment:
This is more expensive than `ensureCapacity` as `ensureCapacity` only adds segments. Can it be slower sometimes in practice? Multi channel still has `io.trino.operator.MultiChannelGroupByHash#groupAddressByGroupId`